Lib/test/test_tempfile.py

```diff
@@ -561,6 +561,23 @@ def test_usable_template(self):
             os.rmdir(d)
 
 
+class TestSanitizeParamsInner(BaseTestCase):
+    """Test the internal function _sanitize_params."""
+
+    def test_throw_exception_on_path_separator_detection(self):
+        with self.assertRaises(ValueError):
+            tempfile.mkstemp(prefix=f"{os.sep}home")
+
+    def test_throw_exception_on_encoded_path_separator_detection(self):
+        with self.assertRaises(ValueError):
+            tempfile.mkstemp(prefix=f"{os.fsencode(os.sep)}home")
+
+    @unittest.skipIf(os.altsep is None, "os.altsep is not present on this platform")
+    def test_throw_exception_on_alternative_path_separator_detection(self):
+        with self.assertRaises(ValueError):
+            tempfile.mkstemp(prefix=f"{os.altsep}home")
```

Comment:
You need also a test for bytes prefix with altsep.

Add also tests for suffix containing a pathname components separator.

Add also tests for other functions that use `_sanitize_params()`.

It will be too wasteful to add 24 separate test methods, so you should test several cases per method method.